draft-packages/avatar-group/KaizenDraft/AvatarGroup/AvatarGroup.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react"
+import cx from "classnames"
+import { v4 } from "uuid"
+import {
+  Avatar,
+  GenericAvatarProps,
+  CompanyAvatarProps,
+} from "@kaizen/draft-avatar"
+import styles from "./styles.module.scss"
+
+type AvatarProps =
+  | Omit<GenericAvatarProps, "size">
+  | Omit<CompanyAvatarProps, "size">
+
+export type AvatarGroupSize = "small" | "medium" | "large"
+export interface AvatarGroupProps {
+  /**
+   * There are 3 fixed sizes available for the AvatarGroup. `"small"` will remove border and box shadow to save space.
+   * @default "medium"
+   */
+  size?: AvatarGroupSize
+  /**
+   * Limits the number of avatars shown within a group. If the number of avatars exceeds the `maxVisible` a counter token will display the remaining avatars not visible.
+   * @default 2
+   */
+  maxVisible: 2 | 3 | 4 | 5 | 6
+  /**
+   * Takes a array of Avatars. Defaults to an empty array
+   * @default []
+   */
+  avatars: AvatarProps[]
+}
+
+const renderAvatars = (
+  avatars: AvatarProps[],
+  size: AvatarGroupSize,
+  maxVisible: number
+) =>
+  avatars?.map(
+    (avatarProps, index) =>
+      index < maxVisible && (
+        <li
+          key={`avatar-${v4()}`}
+          className={styles.AvatarGroupItem}
+          role="presentation"
+        >
+          <Avatar {...avatarProps} size={size} />
+        </li>
+      )
+  )
+
+const renderCounter = (totalAvatars: number, maxVisible: number) =>
+  totalAvatars > maxVisible && (
+    <li
+      aria-label={`There are ${
+        totalAvatars - maxVisible
+      } other memebers of this group`}
+    >
+      <span className={styles.AvatarCounter} aria-hidden="true">
+        +{totalAvatars - maxVisible}
+      </span>
+    </li>
+  )
+
+export const AvatarGroup = ({
+  size = "medium",
+  maxVisible = 2,
+  avatars = [],
+}: AvatarGroupProps) => (
+  <ul
+    className={cx(styles.AvatarGroup, styles[size])}
+    aria-label="Avatar Group"
+  >
+    {renderAvatars(avatars, size, maxVisible)}
```

Comment:
Should there be a check to see whether avatar is not empty and then do the two render functions? It currently renders and empty `<ul>` list when avatar prop is empty. 

..I thought there might be a way to check for an non-empty array prop in typescript...I just did a quick google and found this??

```
type NonEmptyArray<T> = [T, ...T[]];
avatars: NonEmptyArray<AvatarProps>
```

Reply:
Or even if you just guard it with a check before it renders another test would be good.

Reply:
Solid find, I went with this methodology to make it a little more readable without the generic:

avatars: [AvatarProps, ...AvatarProps[]]